Doc update (#203)
ufs-community/ufs-mrweather-app: docs/UsersGuide/source/testing.rst

Testing
=======

> There are around 38 test cases available for the regression testing to ensure the system is installed correctly and works properly. The regression testing (RT) also makes sure that new code upgrades should not have side effects on the existing functionalities. It ensures that the system still works once the latest code changes are done. The RT can be run on Cheyenne, Orion, and Stampede. There is no preexist baseline and the users need to create the baseline by themselves.

Author: Changed, thanks!

> `create_test <https://esmci.github.io/cime/versions/ufs_release_v1.1/html/Tools_user/create_test.html>`_ is the CIME tool used to do the RT.

Author: Changed, thanks!

It can be used as an easy way to run a single basic test or an entire suite of tests.  
`create_test <https://esmci.github.io/cime/versions/ufs_release_v1.1/html/Tools_user/create_test.html>`_ runs a test suite in parallel for improved performance.  
It is the driver behind the automated nightly testing of cime-driven models.

More information about CIME testing can be found on `CIME: Testing <https://esmci.github.io/cime/versions/ufs_release_v1.1/html/users_guide/testing.html>`_.

Test requirements
=================

> In order to run the tests, NCEPLIB and NCEPLIBS-external need to be installed (see :numref:`Chapter %s <config_new_platform>` for instructions). These libraries have been preinstalled on Cheyenne, but not on Stampede and Orion.

Reviewer: ```suggestion
In order to run the tests, NCEPLIBS and NCEPLIBS-external need to be installed (see :numref:`Chapter %s <config_new_platform>` for instructions). These libraries have been preinstalled on Cheyenne, but not on Stampede and Orion.
```

Author: Changed, thanks!

> The input data needed for the tests are staged on preconfigured machine Cheyenne. On Orion and Stampede, data must be acquired from the ftp site and staged on disk (see :numref:`Chapter %s <inputs_and_outputs>`). Then setting the running environment according to :numref:`Chapter %s <quickstart>`. 

Reviewer: We are going to need more specific instructions here. I will provide a suggestion separately.

```suggestion
The input data needed for the tests needs to be on disk before the tests are submitted. It is already staged on Cheyenne since it is a preconfigured platform. On Orion and Stampede, data must be acquired from the ftp site and staged on disk (see :numref:`Chapter %s <inputs_and_outputs>`) before proceeding with the test. Then setting the running environment according to :numref:`Chapter %s <quickstart>`. 
```

Author: Changed.

Testing
=======

There are several tests available as part of the regression testing suite to ensure the system is installed correctly and works properly. The regression test also confirms that code upgrades do not have side effects on existing functionalities and ensures that the system still works after code changes are made. The regression test can only be run on Cheyenne, Orion, and Stampede. 

> Pre-existing baselines are not provided for this App. Users can run the tests without using a baseline (just to certify that the tests run to completeion) or create their own baseline (to compare future runs against).

Reviewer: ```suggestion
Pre-existing baselines are not provided for this App. Users can run the tests without using a baseline (just to certify that the tests run to completion) or create their own baseline (to compare future runs against).
```

Author: Changed, thanks!

`create_test <https://esmci.github.io/cime/versions/ufs_release_v1.1/html/Tools_user/create_test.html>`_ is the CIME tool used to execute the regression tests.
It can be used as an easy way to run a single basic test or an entire suite of tests.  
`create_test <https://esmci.github.io/cime/versions/ufs_release_v1.1/html/Tools_user/create_test.html>`_ runs a test suite in parallel for improved performance.  
It is the driver behind the automated nightly testing of cime-driven models.

More information about CIME testing can be found on `CIME: Testing <https://esmci.github.io/cime/versions/ufs_release_v1.1/html/users_guide/testing.html>`_.

Test requirements
=================
In order to run the tests, NCEPLIBS and NCEPLIBS-external need to be installed (see :numref:`Chapter %s <config_new_platform>` for instructions). These libraries have been preinstalled on Cheyenne, but not on Stampede and Orion.

> The input data needed for the tests needs to be on disk before the tests are submitted. It is already staged on Cheyenne since it is a preconfigured platform. On Orion and Stampede, data must be acquired from the ftp site and staged on disk (see :numref:`Chapter %s <inputs_and_outputs>`) before proceeding with the test. Then setting the running environment according to :numref:`Chapter %s <quickstart>`. 

Reviewer: Avoid the word need twice in same sentence
```suggestion
The input data required for the tests needs to be on disk before the tests are submitted. It is already staged on Cheyenne since it is a preconfigured platform. On Orion and Stampede, data must be acquired from the ftp site and staged on disk (see :numref:`Chapter %s <inputs_and_outputs>`) before proceeding with the test. Then setting the running environment according to :numref:`Chapter %s <quickstart>`. 
```

Author: Changed.

=============================

> **$CIMEROOT/scripts/query_testlists** gathers descriptions of the tests and testlists available

Reviewer: Need to define testlists.

Author: Added the testlist for Cheyenne, thanks!

=============================

**$CIMEROOT/scripts/query_testlists** gathers descriptions of the tests and testlists available

> for UFS, the components, and projects. The available testlists for Cheyenne: ::

Reviewer: ```suggestion
for UFS, the components, and projects. The available tests for Cheyenne are: ::
```
After the green box, explain what testlists are. For example: The results indicate that there are tests available on Cheyenne for two compilers (Intel and GNU). Furthermore, the results indicate that all tests are part of a ``testlist`` called  _prealpha_. ``Testlists`` are lists that aggregate a number of tests under a single umbrella. All tests contained in a ``testlist`` can be run with single command when the ``testlist`` is passed as an argument to the ``create_test`` command.

Author: Changed.

The ``--show-options`` argument does the same, but displays the 'options' defined for the tests,
such as queue, walltime, etc..

Using **create_test** 
==============================

> To run a Smoke startup test::

Reviewer: ```suggestion
To run a SMS test::
```

Author: Changed, thanks!

This will run entire test suite on specified machine ``MACHINE`` such as Stampede2 and generates the baseline under ``BASELINE_ROOT`` directory with a name of ``GENERATE``. 

> Here are the commands to run RT on Cheyenne, Orion, and Stampde::

Reviewer: ```suggestion
The commands to run the regression test on Cheyenne, Orion, and Stampede are below. You must replace the compute projects listed (using variable ``PROJECT``) to a project you can use to run the tests. ::
```
There should not be words inside the code blocks, only commands. So please separate this blurb in three parts, with three code blocks.

Explain why there are 3 sets for Stampede. And that the 3 sets of RTs need to be submitted separately on Stampede, that is, only submit the second set once the 1st is done, and only submit the 3rd once the 2nd is done.

Author: Added, thanks!

Reviewer: There should not be words inside the code blocks, only commands. So please separate this blurb in three parts, with three code blocks, one for each platform.

Reviewer: Explain why there are 3 sets for Stampede and that the 3 sets of RTs need to be submitted separately on Stampede, For example: 
On Stampede it is necessary to submit the tests divided in three ``testlists`` (_prealpha_p1_, _pre_alpha_p2_, and _prealpha_p3_ because there is a limit to the number of jobs a user can have in the queue at a given time. Users should submit each set of tests separately, and wait for all tests to finish before submitting the next set.

Author: Changed, thanks!
